src/h5web/App.test.tsx

```diff
@@ -1,23 +1,134 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import {
+  render,
+  RenderResult,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
 import App from './App';
 import MockProvider from './providers/mock/MockProvider';
 
 const DOMAIN = 'bsa_002_000-integrate-sub';
 
-test('renders with root group selected', async () => {
-  const { findByText, findByRole } = render(
+function renderApp(): RenderResult {
+  return render(
     <MockProvider domain={DOMAIN}>
       <App />
     </MockProvider>
   );
+}
+
+test('renders with root group selected', async () => {
+  const { findByText, getByRole } = renderApp();
 
   const noVisText = await findByText('Nothing to visualize');
```

Comment:
To get an element, try one of the `getBy*` methods first. If it doesn't work, it means the element appears asynchronously, so use `await findBy*` instead.

An element can appear asynchronously when the app re-renders because of an asynchronous state change, which can happen because of a `useEffect` or an HTTP request, notably.